pkg/virt-controller/watch/snapshot/snapshot.go

```diff
@@ -647,6 +634,41 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 			} else {
 				updateSnapshotCondition(vmSnapshotCpy, newProgressingCondition(corev1.ConditionFalse, "Source not locked"))
 			}
+
+			online, err := source.Online()
+			if err != nil {
+				return err
+			}
+
+			if online {
+				indications := []snapshotv1.Indication{snapshotv1.VMSnapshotOnlineSnapshotIndication}
+
+				ga, err := source.GuestAgent()
+				if err != nil {
+					return err
+				}
+
+				if ga {
+					indications = append(indications, snapshotv1.VMSnapshotGuestAgentIndication)
+				} else {
+					indications = append(indications, snapshotv1.VMSnapshotNoGuestAgentIndication)
+				}
+
+				vmSnapshotCpy.Status.Indications = indications
+
+				frozen, err := source.Frozen()
```

Comment:
using this state as an indication for the source state is not so accurate since the status is updated only every 5 secs so it can be already frozen and this will return false, so the frozen condition might never be updated to true. I added those conditions to use to know if unfreeze is needed and if happened it is maybe redundant with this repactor

Reply:
Understood.  Not sure it's a big deal though